core/server/data/xml/sitemap/base-generator.js

```diff
@@ -55,13 +55,18 @@ _.extend(BaseSiteMapGenerator.prototype, {
         // Create all the url elements in JSON
         var self = this,
             nodes;
-        nodes = _.map(data, function (datum) {
+
+        nodes = _.reduce(data, function (nodeArr, datum) {
```

Comment:
The changes to `getData` are an optimisation - we don't fetch data we know won't get included.

The method `createUrlNodeFromDatum` is called in two ways:

1. after doing `getData` to do an initial load on boot
2. after an event with a model that was updated

The validations ensure that we always filter the data correctly no matter how it is fetched (event or API call).

By adding the validations, we change the method `createUrlNodeFromDatum` so that it can return false, and both places where this is called have been updated.